src/main/java/duke/command/AddDeadlineCommand.java

```diff
@@ -0,0 +1,31 @@
+package duke.command;
+
+import duke.TaskList;
+import duke.Ui;
+import duke.Storage;
+import duke.exception.DukeException;
+import duke.task.Deadline;
+
+public class AddDeadlineCommand extends Command {
+    public AddDeadlineCommand(String[] description) {
+        super(description);
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        if (super.description.length < 2) {
+            throw new DukeException("☹ OOPS!!! The description of an event cannot be empty.");
```

Comment:
Hi, since this exception is in relation to user input, I felt it might be more apt if you catch it at the earliest possible point, which is at Parser itself. This applies to other subclass of your Task as well. Again, it is good habit to catch the exception where you throw it, even though the IDE may sometimes offer suggestions for you to add the "throws" clause in the method signature and ask you to catch them where you call the method.